core/domain/exp_jobs_one_off.py

```diff
@@ -791,3 +791,49 @@ def reduce(key, values):
         # Combine all values from multiple lists into a single list
         # for that error type.
         yield (key, list(set().union(*final_values)))
+
+
+class TranslatorToVoiceArtistOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for migrating translator_ids to voice_artist_ids."""
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationRightsModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        translator_ids = item.translator_ids
+        commit_message = 'Migrate from translator to voice artist'
+        commit_cmds = [{
+            'cmd': 'change_role',
+            'assignee_id': translator_id,
+            'old_role': 'translator',
+            'new_role': 'voice artist'
+        } for translator_id in translator_ids]
+
+        if len(translator_ids) > 0:
+            exp_summary_model = exp_models.ExpSummaryModel.get_by_id(item.id)
+
+            if exp_summary_model is None or exp_summary_model.deleted:
```

Comment:
Wait, this seems weird. If this is an error case then we shouldn't really proceed right? We should have a separate audit job that checks this condition, then we run that job and fix any errors, and then only once it succeeds we should run this job (and this job should raise noisily and do nothing for that model if it detects this error).

Reply:
Hmm so it should be a mistake if the rights_model exists but summary_model does not? If this happens, should the one-off job generate a summary model for that exploration?

Reply:
You know, that's a good point. On second thoughts I think what you have now is probably fine? At least at the end of the operation, things are no worse than they were before.

The approach I suggested is the more "careful" one -- to make sure that this situation doesn't even happen in the first place -- but I'm OK with what you have, too.

Reply:
Though you still should try to get rid of the repetition in the code... see @DubeySandeep's comment below.

Reply:
I have done that one but forgot to respond to his comment. 
`item.voice_artist_ids = translator_ids`
`item.translator_ids = []`
`item.commit('Admin', commit_message, commit_cmds)`
These lines have to appear twice because they must happen just before yield and after immigrating to summary models.